tough/src/editor/signed.rs

```diff
@@ -86,7 +87,13 @@ where
             sig: sig.into(),
         });
 
-        // Serialize the newly signed role, and calculate its length and
+        SignedRole::from_signed(role)
+    }
+
+    /// Creates a `SignedRole<Role>` from a `Signed<Role>`.
+    /// This is used to create signed roles for any signed metadata
+    pub fn from_signed(role: Signed<T>) -> Result<SignedRole<T>> {
```

Comment:
Should this be public?  It's a bit confusing unless you knwo the difference between `SignedRole<>` and `Signed<>`